src/main/webapp/resources/js/pages/projects/samples-metadata-import/components/SampleMetadataImportReview.tsx

```diff
@@ -204,28 +208,36 @@ export function SampleMetadataImportReview(): JSX.Element {
       .map((metadataItem) => metadataItem.rowKey);
 
     if (projectId) {
-      const response = await dispatch(
-        saveMetadata({ projectId, selectedMetadataKeys })
-      ).unwrap();
-
-      if (
-        Object.entries(response.metadataSaveDetails).filter(
-          ([, metadataSaveDetailsItem]) => metadataSaveDetailsItem.error
-        ).length === 0
-      ) {
-        navigate(`/${projectId}/sample-metadata/upload/complete`);
-      } else {
-        setLoading(false);
-      }
+      await dispatch(saveMetadata({ projectId, selectedMetadataKeys }))
+        .unwrap()
+        .then(({ metadataSaveDetails }) => {
+          if (
+            Object.entries(metadataSaveDetails).filter(
+              ([, metadataSaveDetailsItem]) => metadataSaveDetailsItem.error
+            ).length === 0
+          ) {
+            navigate(`/${projectId}/sample-metadata/upload/complete`);
+          } else {
+            setLoading(false);
+          }
+        })
+        .catch((payload) => {
```

Comment:
What exactly would this catch? From looking at saveMetadata it should catch everything.

Reply:
One of the test cases I wrote checks if the user has access to the project. If the user no longer has any or inadequate access, an error is thrown when trying to create the metadata fields. This catches the error thrown from line 107 in importReducer.ts and displays the error to the user. Looking at it now, I'm wondering if I should return a rejected status from the thunk instead of throwing an error.